src/Runtime/jni/jnilog.c

```diff
@@ -168,8 +168,8 @@ void log_printf(
    * of 8 bytes on z/OS.
    */
   snprintf(buf + strlen(buf), LOG_MAX_LEN - strlen(buf),
-      "[%016lx][%s]%s:%s:%d ", get_threadid(), log_level_name[level],
-      get_filename(file), func, line);
+      "[%016lx][%s]%s:%s:%d ", (unsigned long)get_threadid(),
+      log_level_name[level], get_filename(file), func, line);
 
```

Comment:
Why is get_threadid() casted to unsigned long? As the comment says, this will not compile on z/OS.